### ISSUE TITLE
Type Issues

### ISSUE THREAD

**Issue by vorticalbox** [davidmdm/myzod#36]
Today I upgraded to 1.6.0 but I am now getting type issues.

```
export const userSchema = myzod.object({
  _id: myzod.string()
    .withPredicate(Types.ObjectId.isValid, '_id must be an object ID')
    .map((id:string) => Types.ObjectId(id)),
  userName: myzod.string(),
});
```

![image](https://user-images.githubusercontent.com/10886065/111314000-e8729700-8658-11eb-9fe6-7cc1e4f0dd06.png)

**Comment by davidmdm** [comment]
@vorticalbox Investigating now!

**Comment by vorticalbox** [comment]
> @vorticalbox Investigating now!

awesome thank you :)

**Comment by davidmdm** [comment]
@vorticalbox Thanks for raising this issue. I pushed up version 1.6.1 which should resolve it. It was a big oversight on my side to not have a test that included a MappedType as part of an objectSchema. I will probably refactor and release a better solution in the coming days but I got something that works around the issue and still presents to intelisense as a `MappedType` while maintaing correct behaviour.

What v1.6+ brings to the table is the ability to add predicate functions and defaults to mapped types. 

Let me k now if v.1.6.1 fixes your issue and I will close this issue :)
